### PR TITLE
Add basic docs and build them in CI

### PR DESCRIPTION
**Pull request by xylar**
This merge adds some basic capabilities to build and deploy the docs in CI.

It also makes a few changes to the design-doc template so it is more compatible with sphinx's table of contents.

It also includes some basic info on how to build the docs locally and what should be included when adding to the docs.

Checklist
* [x] Documentation:
  * [x] Design document has been generated and added to the docs
  * [x] User's Guide has been updated
  * [x] Developer's Guide has been updated
  * [x] Documentation has been [built locally](https://e3sm-project.github.io/omega/develop/developers_guide/building_docs.html) and changes look as expected

### CODE REVIEW

File: components/omega/doc/design/OmegaDesignTemplate.md [hunk 1]
```diff
@@ -1,5 +1,5 @@
-# OMEGA Requirements and Design:
-## *MyClassOrModuleName*
+(omega-dsgn-my-class)=
+# Template: MyClassOrModuleName
```
Comment (xylar) [inline components/omega/doc/design/OmegaDesignTemplate.md:2]: @philipwjones, please let me know if you're okay with the changes to the design-doc template.  The existing title formatting isn't working well in the docs:
image

I'll clean up the existing design docs in another PR.

File: components/omega/doc/index.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+Summer 2026.
+
+The model is written in c++ using the [YAKL](https://github.com/mrnorman/YAKL)
+framework for performance portability.  OMEGA is based on the 
+[TRSK formulation](https://doi.org/10.1016/j.jcp.2009.08.006) for geophysical 
+models on unstructured meshes. The first version of OMEGA will be a direct port
+of the [MPAS-Ocean](https://e3sm.org/model/e3sm-model-description/v1-description/v1-ocean-sea-ice-land-ice/)
+component of E3SM for comparison purposes.
+
+Development is taking place at https://github.com/E3SM-Project/Omega.
```
Comment (xylar) [inline components/omega/doc/index.md:16]: @vanroekel, this is the main part I want your review on.  Would you like to flesh this out further (now or later)?  Anything here you're not happy with?

File: components/omega/doc/design/OmegaDesignTemplate.md (hunk 1 above)
Comment (xylar) [reply components/omega/doc/design/OmegaDesignTemplate.md]: Here's what that looks like with the clean-up in #15:
image

**Comment by xylar** [comment]
I'm going to add a page to the developer's guide with instructions for building and viewing the documentation locally.

File: components/omega/doc/devGuide/Docs.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+- Flags related to the feature that a user can (or must) set when building
+  OMEGA,
+- Config options that a user can modify related to the feature
+
+Documentation for the feature in the Developer's Guide may contain:
+
+- Relevant technical details about development specific to that feature
+- Details on unit tests that cover that feature
+- Details on polaris regression tests that cover that feature
```
Comment (xylar) [inline components/omega/doc/devGuide/Docs.md:57]: @vanroekel and @philipwjones, this is by no means a complete or exhaustive list.  Feel free to suggests changes and additions either now or later when we have more experience and more examples to work from.

Comment (philipwjones) [reply components/omega/doc/devGuide/Docs.md:57]: User guide scope is about right - might need at least a brief explanation of a feature to know what the config options mean.

The developer guide should also serve as a reference manual, so should describe not only the actual mathematical terms (or a reference to them) but also the discretized form with variables close to what are used in the code. This provides an easy way for other developers or reviewers to verify the code against the intended implementation

**Review by philipwjones — APPROVED** [review]
Thanks @xylar . These look great and I agree with the design template changes - obvious in retrospect that the titles shouldn't be generic when integrating.

File: components/omega/doc/devGuide/Docs.md (hunk 3 above)
Comment (xylar) [reply components/omega/doc/devGuide/Docs.md:57]: > might need at least a brief explanation of a feature to know what the config options mean.

I'm not sure what term you want to use.  These are what would be namelist options for a Fortran code but are now in the YAML file.

Comment (xylar) [reply components/omega/doc/devGuide/Docs.md:57]: Your suggestions for the developer's guide are great!

**Comment by xylar** [comment]
@philipwjones thanks for the review! I updated the description of what goes into the Developer's Guide (and Reference Manual) as you suggested.

File: components/omega/doc/devGuide/Docs.md (hunk 3 above)
Comment (xylar) [reply components/omega/doc/devGuide/Docs.md]: Ah, I see. I'll add that. I completely agree.

Comment (xylar) [reply components/omega/doc/devGuide/Docs.md]: Done.

File: components/omega/doc/index.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
+(omega-home)=
+# OMEGA
+
+The Ocean Model for E3SM Global Applications (OMEGA) is an eddy-resolving,
```
Comment (vanroekel) [inline components/omega/doc/index.md:4]:
```suggestion
The Ocean Model for E3SM Global Applications (OMEGA) initially is an eddy-resolving, 
```

File: components/omega/doc/index.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
+(omega-home)=
+# OMEGA
+
+The Ocean Model for E3SM Global Applications (OMEGA) is an eddy-resolving, 
+global ocean model in the early stages of development by the 
+[E3SM](https://e3sm.org/) ocean team.  The first release is planned for
+Summer 2026.
```
Comment (vanroekel) [inline components/omega/doc/index.md:7]:
```suggestion
Summer 2026.  A non-eddying configuration will be released in early 2027.
```

File: components/omega/doc/index.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+The model is written in c++ using the [YAKL](https://github.com/mrnorman/YAKL)
+framework for performance portability.  OMEGA is based on the 
+[TRSK formulation](https://doi.org/10.1016/j.jcp.2009.08.006) for geophysical 
+models on unstructured meshes. The first version of OMEGA will be a direct port
```
Comment (vanroekel) [inline components/omega/doc/index.md:12]:
```suggestion
models on unstructured meshes. The first version of OMEGA will primarily be a direct port
```

File: components/omega/doc/index.md (hunk 2 above)
Comment (vanroekel) [reply components/omega/doc/index.md:16]: Thanks for pointing me to this, I think it looks great.  I made a few minor tweaks.

**Review by vanroekel — COMMENTED** [review]
@xylar I took a quick look through everything and a closer look at the section you flagged me on.  i think things look great!  I only had a few minor tweaks to suggest.

**Comment by xylar** [comment]
Thanks @vanroekel!  I made your suggested changes.

@philipwjones, I think this is ready to merge when you like.  I don't believe I have merging privileges (which is appropriate).

**Comment by xylar** [comment]
Thanks @philipwjones.

**Pull request merged**